ngraph/CMakeLists.txt

```diff
@@ -481,11 +483,13 @@ if (NGRAPH_EXPORT_TARGETS_ENABLE)
     set(NGRAPH_TARGETS_FILE "${CMAKE_CURRENT_BINARY_DIR}/ngraphTargets.cmake")
     export(TARGETS ngraph NAMESPACE ngraph:: FILE "${NGRAPH_TARGETS_FILE}")
 
-    install(EXPORT ngraphTargets
-        FILE ngraphTargets.cmake
-        NAMESPACE ngraph::
-        DESTINATION ${NGRAPH_COMPONENT_PREFIX}cmake
-        COMPONENT ngraph)
+    if(NOT NGRAPH_STATIC_LIBRARY)
+        install(EXPORT ngraphTargets
+            FILE ngraphTargets.cmake
+            NAMESPACE ngraph::
+            DESTINATION ${NGRAPH_COMPONENT_PREFIX}cmake
+            COMPONENT ngraph)
+    endif()
```

Comment:
Because I wrote this thinking about the first step where we deliver inference_engine.dll embeding ngraph as a static library. But you are right, in the end if everything is static we'll need to install all the static libraries.

Reply:
Ok, suggest leaving this place as is and fix later once we have the whole inference_engine.a with ngraph and all other libraries. I think in that stage we will have better understanding how we can export our targets.

Reply:
Maybe this PR should aim at implementing the whole static library to avoid extra work for intermediary steps

Reply:
Up to you. If you are ready to implement next steps, let's do it in this PR by adding new commits.

Reply:
@emmanuelattia-philips do you plan to make more libraries to be static in this PR? Or we can merge once CI has passed

Reply:
@emmanuelattia-philips do you plan to make more libraries to be static in this PR? Or we can merge once CI has passed

Reply:
I'm working on this in the background so i'm a bit slow, i'm currently making sure that this branch and master produces exactly the same number of shared libs with the default CMake command.

Reply:
ok, sounds good, thanks!

Reply:
I think it's about right now. It does exactly what the PR describe, i'll move on in another PR.

Reply:
Do you know if it will be merged to know if i go on in another PR for the rest of the "statification"

Reply:
we just need green precommit and the same number of shared libraries as before. Other improvements can be done later

Reply:
I checked there is the same number of shared libs as before, and for the  precommit i don't have access to ci/jenkins so i don't know what is blocking

Reply:
Could you please merge with latest master and push? CI will run automatically.

Reply:
I did that, still no luck with "ci/jenkins"

Reply:
I've rerun to see all errors. I will let you know

Reply:
Cross-compilation for ARM fails with:
```
[2020-11-09T17:56:45.094Z] CMake Warning at cmake/shellcheck/shellcheck.cmake:11 (message):

[2020-11-09T17:56:45.094Z]   shellcheck tool is not found

[2020-11-09T17:56:45.094Z] Call Stack (most recent call first):

[2020-11-09T17:56:45.094Z]   CMakeLists.txt:136 (ie_shellcheck_process)

[2020-11-09T17:56:45.094Z] 

[2020-11-09T17:56:45.094Z] 

[2020-11-09T17:56:45.094Z] CMake Error: The following variables are used in this project, but they are set to NOTFOUND.

[2020-11-09T17:56:45.094Z] Please set them or make sure they are set and tested correctly in the CMake files:

[2020-11-09T17:56:45.094Z] Protobuf_LIBRARY

[2020-11-09T17:56:45.094Z]     linked by target "onnx_proto" in directory /home/jenkins/agent/workspace/private-ci/ie/build-linux-debian_9_arm/b/build/dldt/_deps/ext_onnx-src

[2020-11-09T17:56:45.094Z] 

[2020-11-09T17:56:45.094Z] -- Configuring incomplete, errors occurred!

[2020-11-09T17:56:45.094Z] See also "/home/jenkins/agent/workspace/private-ci/ie/build-linux-debian_9_arm/b/build/dldt/CMakeFiles/CMakeOutput.log".

[2020-11-09T17:56:45.094Z] See also "/home/jenkins/agent/workspace/private-ci/ie/build-linux-debian_9_arm/b/build/dldt/CMakeFiles/CMakeError.log".
```

Reply:
Is it possible to get the content of the logs ?

Reply:
I have trouble reproducing the issue, i set up a Debian 10 (under WSL) and tried this (disabled myriad because i have no libusb for cross compilation):

rm -fR build && cmake -Bbuild -DCMAKE_TOOLCHAIN_FILE=cmake/arm.toolchain.cmake -DENABLE_MYRIAD=OFF

With success

Reply:
Ok I reproduced it with a Debian 9 (under Virtual Box), that will help

Reply:
cool, because from the logs it's hard to say what happens wrong..

Reply:
Ok so when i scoped into a function the include(cmake/external_protobuf.cmake) to change BUILD_SHARED_LIBS locally, it seems that when cross compiling + some old version of protobuf, this cmake file has some needed border effects. 

So instead of a pretty "include_as_static" I have to do an ugly:

        set(BEFORE_PROTOBUF_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
        set(BUILD_SHARED_LIBS OFF)
        include(cmake/external_protobuf.cmake
        set(BUILD_SHARED_LIBS ${BEFORE_PROTOBUF_BUILD_SHARED_LIBS})
        unset(BEFORE_PROTOBUF_BUILD_SHARED_LIBS)

But at least it now passes. What do you think ?

Reply:
But what is the difference with `include_as_static`? `include_as_static` implemented as a cmake function also minimized the scope of variable..

Reply:
external_protobuf.cmake redefines Protobuf_LIBRARY in some case and i think it's used by external_onnx.cmake, that's if i include it in a scope the cmake scripts fails when processing external_onnx.cmake

By the way ci/jenkins still fails.